### ISSUE TITLE
Is it possible to configure PiHole as a LAN DNS server, where hosts are assigned IPs dynamically using PiHole's DHCP server?

### ISSUE THREAD

**Issue by edbird** [pi-hole/pi-hole#3083]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

1

---
**Expected behaviour:**

Not relevant.

**Actual behaviour:**

I don't think this is implemented yet, although it may be possible by editing config files?

**Steps to reproduce:**

N/A

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

N/A

**Troubleshooting undertaken, and/or other relevant information:**

N/A

Title is fairly self explanatory, however here is a more in-depth explanation.

I have configured PiHole to be the DHCP server for my local network. It appears to be working and is assigning IPs dynamically to hosts.

I have several devices on this network, each with a unique hostname.

I would like to be able to configure Pi-Hole so that it acts as a DNS server for these local hostnames.

Is this possible?

If it is possible, does this require that I configure my devices with static IP addresses, or can this be done in conjunction with DHCP?

Possibly such a thing would require me to write some new software which sends hostnames to PiHole after an IP is obtained via DHCP. I could possibly do such a thing, although it may take me a while to figure out how.

Regarding expected behaviour, from the computer I am currently using, I would anticipate being able to do 

```
ssh username@pc2020
or possibly
ssh username@pc2020.mydomain
```

**Comment by edbird** [comment]
Update: It appears to already be working? (Sort of.)

One of my hosts is accessible using `ssh user@pcname.piholelan` - `piholelan` is the domain name set under "Advanced DHCP settings".

However the other device currently connected, as well as the device running PiHole itself do not work.

I should be able to do:

```
ssh pi@rpi1.piholelan
and
ssh username@server.piholelan
```

however this does not work

**Comment by Mograine** [comment]
Hello,

If you use piHole as your DNS you can set custom hosts in the "Custom DNS" section.

You can for example add server.local/192.168.0.250 in that section and you'll be able to access your server by typing server.local from now on

**Comment by edbird** [comment]
@Mograine that sounds about right - where is the custom DNS section? Is that part of the web interface?

**Comment by Mograine** [comment]
Yes my bad, i thought we were on the AdminLTE repository x)

You can use the two integrated commands :

pihole -a addcustomdns 192.168.0.250 server.local
pihole -a removecustomdns 192.168.0.250 server.local

**Comment by edbird** [comment]
@Mograine Hi and sorry for the further questions, however - 

1: Is there an associated config file, where I can put these custom DNS addresses in? In other words, where do these commands store the associated data, and can one edit those files directly?

2: Will this be implemented in the web GUI?

3: What is the AdminLTE repository?

**Comment by Mograine** [comment]
@edbird After checking, i noticed that this functionality is not yet delivered in stable version of PiHole, my code was accepted in november and last stable release is from septembre. You won't be able to use it at the moment. Check point 1 to do it manually atm.

1: You can find the custom DNS addresses in /etc/pihole/custom.list , these commands only handle add/delete in this file for you (and reloading DNS pi-hole server). Which also mean that while you wait for the new stable version, you can manually create this file and add lines with format "ip domain", then you can reload with "pihole -a restartdns"

2: It is implemented, should be available in the next stable release as well

3: Pi-hole has two Github repository, this one for the "core" of the system (DNS servers, DHCP, bash scripts, etc) and the "https://github.com/pi-hole/AdminLTE" repository were the web GUI is stored and developped.

**Comment by edbird** [comment]
@Mograine Ok thanks for the info - looks like `addcustomdns` isn't yet available in the current release. Any idea approximately how long it will be before this reaches the next release, or is this something you don't have control over?

**Comment by edbird** [comment]
@Mograine Followed your instructions to add manually, and successfully restarted the DNS service, however it does not appear to be working. Do you have any advice on what I might do to debug this?

**Comment by Mograine** [comment]
I sadly don't have control over the next release date, i'm just a contributor to this project not a member :)

I forgot to tell you to add a line in the file "advanced/01-pihole.conf"

after the line "addn-hosts=/etc/pihole/local.list"
you should add "addn-hosts=/etc/pihole/custom.list"

**Comment by Mograine** [comment]
This issue can be closed as PR #2978 have been merged in the newer 5.0 release, you can now use advantage of the "Local DNS Records" functionnality
